crates/fiber-lib/src/cch/actor.rs

```diff
@@ -482,11 +458,47 @@ impl CchState {
         Ok(())
     }
 
-    async fn receive_btc(
+    async fn handle_invoice_changed_event(
         &mut self,
-        myself: ActorRef<CchMessage>,
-        receive_btc: ReceiveBTC,
-    ) -> Result<CchOrder, CchError> {
+        mut order: CchOrder,
+        status: CchIncomingPaymentStatus,
+    ) -> Result<()> {
+        // When the order is from fiber to lightning, the invoice is created in fiber. Since Fiber
+        // events are handled in `handle_pending_received_tlc_notification` and
+        // `handle_settled_tlc_notification`, skip the event here.
+        if order.is_from_fiber_to_lightning() {
+            return Ok(());
+        }
+
+        order.status = status.into();
+        if status == CchIncomingPaymentStatus::Accepted {
+            let message = |rpc_reply| -> NetworkActorMessage {
+                NetworkActorMessage::Command(NetworkActorCommand::SendPayment(
+                    SendPaymentCommand {
+                        invoice: Some(order.outgoing_pay_req.clone()),
+                        ..Default::default()
+                    },
+                    rpc_reply,
+                ))
+            };
+
+            let payment_status = call!(self.network_actor, message)
```

Comment:
prefer to use `self.network_actor.send_message(..)` since most of current code are following this style.